### PR TITLE
Increase and optimize HUD update speed, add advanced speed color options, track jumpbugs in HUD

### LINKED ISSUE #224: Different prespeed color for jumpbugs/teleperf
**[Original report](https://bitbucket.org/kztimerglobalteam/gokz/issue/224) by zer0.k (Bitbucket: [zer0k_z](https://bitbucket.org/zer0k_z), ).**

----------------------------------------

This is mostly a vanilla QoL change, right now the speed indicator does not differentiate between jumpbugs and normal perfs.

### PR DESCRIPTION
**Pull request by zer0k-z**
Partially resolves #224

### CODE REVIEW

**Comment by zealain** [comment]
I like the speed color changes, I think we can show the special jumpbug color even without the advanced option enabled and only put the airaccel thingy behind that option. Which I would then also rename a bit to make it more clear what it does.

Don't like the update rate changes though, all it does is make it faster and didn't really help when lagging either. The current update rate is deliberately not chosen to be higher to avoid a jittery feel to it.

**Comment by zer0k-z** [comment]
The update rate change was meant to make full use of a player's good connection to the server with low ping. Its behavior should not change when the player has high ping at all.

For the jumpbug color, it is optional because a jumpbug is almost the exact same as a perf on kzt/skz. But I guess it wouldn't hurt knowing that either.

**Comment by zer0k-z** [comment]
I forgot that there are two different changes to the update speed, depending on which one you use.

1. For the HUD text: This is just an improvement in all cases, more responsiveness is just better in all network scenarios. 21Hz update rate feels sluggish, especially compared to MovementHUD's 128Hz. Doubling it will be a very noticeable improvement while not reaching the point of diminishing returns and I doubt there is any downside at all to this.

2. For the center panel: The center panel update speed is painfully slow, 10x slower than MHUD. This discourages a lot of people from ever using this, which I think it's a shame. 

But the thing with center panel is that its maximum update speed depends on the ping between the player and the server. Right now it's the equivalent of player having 78ms ping to the server, but if they have a good connection to the server. I don't see why we shouldn't give them the best experience possible.

**Comment by zealain** [comment]
As I already outlined before a faster update speed is **not** just a flat out improvement. It can make it more jittery and harder to read, making it just a matter of preference. As of right now, the update rate will not be increased. There is a widely available higher update rate alternative with mhud that people who prefer that experience can already use.

**Comment by zer0k-z** [comment]
In that case, what about an option for higher update rate?

**Comment by zealain** [comment]
An option is fine

File: addons/sourcemod/scripting/gokz-hud/racing_text.sp [hunk 1]
```diff
@@ -21,7 +21,8 @@ void OnPluginStart_RacingText()
 
 void OnPlayerRunCmdPost_RacingText(int client, int cmdnum)
 {
-	if (gB_GOKZRacing && cmdnum % 6 == 3)
+	int updateSpeed = GOKZ_HUD_GetOption(client, HUDOption_UpdateRate) == UpdateRate_Fast ? 3 : 6;
```
Comment (zealain) [inline addons/sourcemod/scripting/gokz-hud/racing_text.sp:24]: Same as for the center panel

File: addons/sourcemod/scripting/gokz-hud/speed_text.sp [hunk 2]
```diff
@@ -24,7 +24,8 @@ void OnPluginStart_SpeedText()
 
 void OnPlayerRunCmdPost_SpeedText(int client, int cmdnum, HUDInfo info)
 {
-	if (cmdnum % 6 == 0 || info.IsTakeoff)
+	int updateSpeed = GOKZ_HUD_GetOption(client, HUDOption_UpdateRate) == UpdateRate_Fast ? 3 : 6;
```
Comment (zealain) [inline addons/sourcemod/scripting/gokz-hud/speed_text.sp:27]: Same as for the center panel

File: addons/sourcemod/scripting/gokz-hud/timer_text.sp [hunk 3]
```diff
@@ -43,7 +43,8 @@ void OnPluginStart_TimerText()
 
 void OnPlayerRunCmdPost_TimerText(int client, int cmdnum, HUDInfo info)
 {
-	if (cmdnum % 6 == 3)
+	int updateSpeed = GOKZ_HUD_GetOption(client, HUDOption_UpdateRate) == UpdateRate_Fast ? 3 : 6;
```
Comment (zealain) [inline addons/sourcemod/scripting/gokz-hud/timer_text.sp:46]: Same as for center panel

File: addons/sourcemod/scripting/gokz-hud/tp_menu.sp [hunk 4]
```diff
@@ -21,7 +21,8 @@
 
 void OnPlayerRunCmdPost_TPMenu(int client, int cmdnum, HUDInfo info)
 {
-	if (cmdnum % 6 == 3)
+	int updateSpeed = GOKZ_HUD_GetOption(client, HUDOption_UpdateRate) == UpdateRate_Fast ? 3 : 6;
```
Comment (zealain) [inline addons/sourcemod/scripting/gokz-hud/tp_menu.sp:24]: Same as for center panel (man, there are a lot of update rates xD)

File: addons/sourcemod/scripting/include/gokz/hud.inc [hunk 5]
```diff
@@ -190,9 +213,10 @@ stock int gI_HUDOptionDefaults[HUDOPTION_COUNT] =
 	TimerText_InfoPanel, 
 	TimerStyle_Standard, 
 	TimerType_Enabled,
-	SpeedText_InfoPanel, 
+	SpeedText_InfoPanel,
 	ShowWeapon_Enabled,
-	ReplayControls_Enabled
+	ReplayControls_Enabled,
+	DeadstrafeColor_Disabled
```
Comment (zealain) [inline addons/sourcemod/scripting/include/gokz/hud.inc:219]: I believe this is missing a default for the update rate.

**Review by zealain — CHANGES_REQUESTED** [review]

**Review by zealain — APPROVED** [review]

**Pull request merged**